### PR TITLE
Page for showing corrective actions

### PR DESCRIPTION
**Pull request by frankieroberto**
Following on from #668, this adds a new page for viewing "corrective actions" (product recalls, etc). 

This is currently only linked to from the Activity timeline, but will soon be listed and linked to from a "Supporting information" tab, replacing the "Attachments" tab.  Due to existing database model limitations, it is currently only possibly to create the link from the Activity timeline if a file was attached to the corrective action. This can be remedied in a future change.

### Before

Activity timeline entry links only to the attachment:

Screenshot 2020-06-04 at 09 21 40

### After

Activity timeline entry links to the corrective action (only if an attachment was added):

Screenshot 2020-06-04 at 09 21 53

New corrective action page:

Screenshot 2020-06-04 at 09 22 07

### Implementation notes

We have an existing `CorrectiveActionsController` but that uses the `show` action as part of the "add a corrective action" flow. To avoid conflicting with or refactoring that immediately, I’ve added a new `ViewCorrectiveActionsController`, which isn't ideal but is a short-term compromise.

### CODE REVIEW

File: psd-web/app/views/investigations/activities/_activity_card.html.erb [hunk 1]
```diff
@@ -52,6 +52,9 @@
 
     <% if activity.is_a? AuditActivity::Test::Result %>
       <%= link_with_hidden_text_to "View test result", " from #{activity.test_result.date.to_s(:govuk)}", investigation_test_result_path(@investigation, activity.test_result.id) %>
+
+    <% elsif activity.is_a?(AuditActivity::CorrectiveAction::Add) && activity.corrective_action %>
+      <%=  link_to "View corrective action", investigation_action_path(@investigation, activity.corrective_action), class: "govuk-link" %>
```
Comment (frankieroberto) [inline psd-web/app/views/investigations/activities/_activity_card.html.erb:57]: Not ideal, but we should refactor this to use the new `metadata` json object and separate partials per activity entry in a separate PR.

File: psd-web/config/routes.rb [hunk 2]
```diff
@@ -140,6 +140,8 @@
 
     resources :test_results, controller: "investigations/test_results", only: :show, path: "test-results"
 
+    resources :actions, controller: "investigations/view_corrective_actions", only: :show, path: "corrective-actions"
```
Comment (frankieroberto) [inline psd-web/config/routes.rb:143]: Resource uses a shorter name (`:actions` instead of `:corrective_actions`) to avoid conflicting with the existing resource.

File: psd-web/app/views/investigations/corrective_actions/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        key: { text: "Duration of measure" },
+        value: { text: @corrective_action.duration.upcase_first }
+      },
+      {
+        key: { text: "Scope" },
+        value: { text: @corrective_action.geographic_scope }
+      }
+    ]
+
+    if @corrective_action.details.present?
```
Comment (StupidCodeFactory) [inline psd-web/app/views/investigations/corrective_actions/show.html.erb:50]: might be wrong but I can't see a test for this?

Comment (frankieroberto) [reply psd-web/app/views/investigations/corrective_actions/show.html.erb:50]: It's covered in the feature spec: https://github.com/UKGovernmentBEIS/beis-opss-psd/pull/684/files#diff-72f1fd770cc58e1731a1a09a80797d35R68

File: psd-web/app/views/investigations/corrective_actions/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    <div class="app-meta-area">
+      <p class="govuk-body govuk-hint">
+        Added <%= @corrective_action.created_at.to_s(:govuk) %>
+      </p>
+    </div>
+
+    <% rows = [
+      {
+        key: { text: "Date" },
```
Comment (edwardhorsford) [inline psd-web/app/views/investigations/corrective_actions/show.html.erb:25]:
```suggestion
        key: { text: "Date of action" },
```

File: psd-web/app/views/investigations/corrective_actions/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      {
+        key: { text: "Type of action" },
+        value: { text: @corrective_action.measure_type.upcase_first }
+      },
+      {
+        key: { text: "Duration of measure" },
+        value: { text: @corrective_action.duration.upcase_first }
+      },
+      {
+        key: { text: "Scope" },
```
Comment (edwardhorsford) [inline psd-web/app/views/investigations/corrective_actions/show.html.erb:45]:
```suggestion
        key: { text: "Geographic scope" },
```

Comment (frankieroberto) [reply psd-web/app/views/investigations/corrective_actions/show.html.erb:45]: Heh, it was "Scope" in the prototype screenshot. I actually think it's better than "Geographic scope", which feels a bit redundant once you can see the value, and a slightly awkward word?

File: psd-web/app/views/investigations/corrective_actions/show.html.erb (hunk 3 above)
Comment (StupidCodeFactory) [reply psd-web/app/views/investigations/corrective_actions/show.html.erb:50]: Sorry I meant when the details are not present, I saw this was added to the feature test and wondered where the other branch was covered.

File: psd-web/app/views/investigations/corrective_actions/show.html.erb (hunk 5 above)
Comment (edwardhorsford) [reply psd-web/app/views/investigations/corrective_actions/show.html.erb:45]: Ok, persuaded.

File: psd-web/app/views/investigations/corrective_actions/show.html.erb (hunk 3 above)
Comment (frankieroberto) [reply psd-web/app/views/investigations/corrective_actions/show.html.erb:54]: I didn't feel the need to test the row not appearing if there was no data, in this instance. Mainly to avoid the feature specs getting slow.

If this gets any more complicated, we can consider refactoring into a helper and testing that, but for now I think it's simple enough not to need it.
